catalog/dags/common/utils.py

```diff
@@ -0,0 +1,76 @@
+import functools
+from inspect import _ParameterKind, signature
+from typing import Any
+
+from common.constants import SQL_INFO_BY_MEDIA_TYPE
+
+
+def setup_kwargs_for_media_type(
+    values_by_media_type: dict[str, Any], kwarg_name: str
+) -> callable:
+    """
+    Create a decorator which provides media_type-specific information as parameters
+    for the called function. The called function must itself have a media_type kwarg,
+    which is used to select values.
+
+    Required arguments:
+
+    values_by_media_type: A dict mapping media types to arbitrary values, which may
+                          themselves be of any type
+    kwarg_name:           The name of the kwarg that will be passed to the called
+                          function
+
+    Usage example:
+
+    @setup_kwargs_for_media_type(MY_VALS_BY_MEDIA_TYPE, 'foo')
+    def my_fun(media_type, foo = None):
+        ...
+
+    When `my_fun` is called, if the `foo` kwarg is not passed explicitly, it will be set
+    to the value of MY_VALS_BY_MEDIA_TYPE[media_type]. An error is raised for an invalid
+    media type.
+    """
+
+    def wrap(func: callable) -> callable:
+        """
+        Provide the appropriate value for the media_type passed in the called function.
+        If the called function is already explicitly passed a value for `kwarg_name`,
+        use that value instead.
+        """
+
+        # The called function must be supplied a `media_type` keyword-only argument. It
+        # cannot allow the value to be supplied as a positional argument.
+        if (
+            media_type := signature(func).parameters.get("media_type")
+        ) is None or media_type.kind != _ParameterKind.KEYWORD_ONLY:
+            raise Exception(
+                "Improperly configured: `media_type` must be a keyword-only argument"
```

Comment:
Would it be useful for debugging down the road to add the name of what is _improperly configured_ here? It's the wrapped function, right?

Reply:
Yes, that's a great point! Adding this in.